ext/com/concurrent_ruby/ext/SynchronizationLibrary.java

```diff
@@ -55,14 +55,14 @@ public IRubyObject initialize(ThreadContext context, IRubyObject[] args, Block b
             }
         }
 
-        @JRubyMethod(name = "synchronize", visibility = Visibility.PRIVATE)
+        @JRubyMethod(name = "synchronize", visibility = Visibility.PROTECTED)
```

Comment:
@jdantonio it's counter intuitive but private methods can be actually called from sub-classes in Ruby
```ruby
A = Class.new { private def a; :a; end }; B = Class.new(A); B.new.instance_eval {a} # => :a
```
It seemed weird to me too, I should have thought about it more.